src/bin/krill.rs

```diff
@@ -2,14 +2,34 @@ extern crate krill;
 
 use std::sync::Arc;
 
-use krill::daemon::http::server;
+use clap::{App, Arg};
+use log::error;
+
+use krill::{
+    constants::{KRILL_DEFAULT_CONFIG_FILE, KRILL_SERVER_APP, KRILL_VERSION},
+    daemon::{config::Config, http::server},
+};
 
 #[tokio::main]
 async fn main() {
-    match server::parse_config() {
+    let matches = App::new(KRILL_SERVER_APP)
+        .version(KRILL_VERSION)
+        .arg(
+            Arg::with_name("config")
+                .short("c")
+                .long("config")
+                .value_name("FILE")
+                .help("Override the path to the config file (default: './defaults/krill.conf')")
+                .required(false),
+        )
+        .get_matches();
+
+    let config_file = matches.value_of("config").unwrap_or(KRILL_DEFAULT_CONFIG_FILE);
+
+    match Config::create(config_file, false) {
         Ok(config) => {
             if let Err(e) = server::start_krill_daemon(Arc::new(config)).await {
-                eprintln!("Krill failed to start: {}", e);
+                error!("Krill failed to start: {}", e);
```

Comment:
yes, this is done as when the `config` is created. So if we have one, then it's initialised.

Reply:
Hmm, I thought when I looked at `Config::create()` that it seemed to me that it could bail out before the logging was initialized if initial parsing failed? It says `Self::read_config(config_file)?` and then only after that `config.init_logging()?` ??